test/react_client/js_interop_helpers_test.dart

```diff
@@ -131,5 +134,82 @@ main() {
       expect(objectKeys(convertedNestedJsObject), expectedProperties,
           reason: 'JS object should not have any additional properties');
     });
+
+    group('works as expected when the map contains frozen objects:', () {
+      test('anonymous object', () {
+        final frozenAnonymousObject = jsify({'foo': 'bar'});
+        expect(_getPrototypeOf(frozenAnonymousObject), _objectPrototype,
+            reason: 'test setup check; should not extend from another JS type');
+
+        _freeze(frozenAnonymousObject);
+        expect(_isFrozen(frozenAnonymousObject), isTrue, reason: 'test setup check; should have frozen');
+
+        dynamic jsObject;
+        expect(() {
+          jsObject = jsifyAndAllowInterop({
+            'frozenObject': frozenAnonymousObject,
+          });
+        }, returnsNormally, reason: 'should not throw when it encounters a frozen object');
+        final convertedNestedFrozenObject = getProperty(jsObject, 'frozenObject');
+        expect(convertedNestedFrozenObject, same(frozenAnonymousObject),
+            reason: 'JS object should have just gotten passed through');
+      });
+
+      test('non-anonymous object', () {
+        final frozenNonAnynymousObject = Foo(21);
+        expect(_getPrototypeOf(frozenNonAnynymousObject), isNot(_objectPrototype),
+            reason: 'test setup check; should extend from another JS type');
+
+        _freeze(frozenNonAnynymousObject);
+        expect(_isFrozen(frozenNonAnynymousObject), isTrue, reason: 'test setup check; should have frozen');
+
+        dynamic jsObject;
+        expect(() {
+          jsObject = jsifyAndAllowInterop({
+            'frozenObject': frozenNonAnynymousObject,
+          });
+        }, returnsNormally, reason: 'should not throw when it encounters a frozen object');
+        final convertedNestedFrozenObject = getProperty(jsObject, 'frozenObject');
+        expect(convertedNestedFrozenObject, same(frozenNonAnynymousObject),
+            reason: 'JS object should have just gotten passed through');
+      });
+
+      test('array object constructed in JS', () {
+        // Create this and freeze it on the JS side so that we're not starting out with Dart's list implementation
+        // and potentially any wrapper classes.
```

Comment:
```suggestion
        // Create this on the JS side so that we're not starting out with Dart's list implementation
        // and potentially any wrapper classes.
```